packages/plugin-relay/src/types.ts

```diff
@@ -49,20 +49,24 @@ export type RelayPluginOptions<Types extends SchemaTypes> = EmptyToOptional<{
       OutputRefShape<GlobalIDShape<Types> | string>,
       boolean,
       { id: InputFieldRef<InputShape<Types, 'ID'>> },
-      Promise<unknown>
+      Promise<Types['Objects']>
     >,
     'args' | 'resolve' | 'type'
-  >;
+  > & {
+    resolve?: (id: string, context: Types['Context']) => MaybePromise<Types['Objects']>;
+  };
   nodesQueryOptions: Omit<
     PothosSchemaTypes.QueryFieldOptions<
       Types,
       [OutputRefShape<GlobalIDShape<Types> | string>],
       FieldNullability<[unknown]>,
       { ids: InputFieldRef<InputShape<Types, 'ID'>[]> },
-      Promise<unknown>[]
+      Promise<Types['Objects']>[]
     >,
     'args' | 'resolve' | 'type'
-  >;
+  > & {
+    resolve?: (ids: string[], context: Types['Context']) => MaybePromise<Types['Objects'][]>;
```

Comment:
you can remove this, and just remove the `resolve` from the `Omit` above if you take all the normal resolve args

Reply:
i could remove `resolve` from `Omit` but then the resolve options is not nullable so users would have to define it. If people upgrade the package it would break their code.

Reply:
ah right, you probably still want to match the resolve type to be similar to the original resolve method (all 4 args, plus supporting the full variety of supported return types).  You can probably just look at how resolve is implemented on the original options object